src/exact/ExactMapper.cpp

```diff
@@ -662,8 +670,10 @@ void ExactMapper::coreMappingRoutine(const std::set<unsigned short>& qubitChoice
     //////////////////////////////////////////
     /// 	Solving							//
     //////////////////////////////////////////
-    if (sat == opt.check()) {
-        model m               = opt.get_model();
+    lb->produceInstance();
+    Result res = lb->solve();
+    if (Result::SAT == res) {
+        Model* m              = lb->getModel();
```

Comment:
```suggestion
        const auto m              = lb->getModel();
```
It should be possible to make the generated model `const`. However, this requires some functions in `LogicBlocks` to also be `const`. I currently lack the in-depth knowledge of the `LogicBlocks` library to find out whether that is feasible.